### PR TITLE
Adds COUNTER type for read/write.

### PR DESCRIPTION
**Pull request by glcj**
This PR includes the COUNTER type to be processed by the S7 driver.

Reading and writing of the COUNTER type is included, as well as reading it in arrays.

Manual testing against an S7-400, CPU 417 is added.

### CODE REVIEW

File: plc4j/drivers/s7/src/test/java/org/apache/plc4x/java/s7/readwrite/DatatypesTest.java [hunk 1]
```diff
@@ -53,9 +63,12 @@ public static void main(String[] args) throws Exception {
             builder.addTagAddress("udint-array", "%DB2:50:UDINT[2]"); // 12345, 23456
             builder.addTagAddress("real-value", "%DB2:58:REAL"); // 3.14159
             builder.addTagAddress("real-array", "%DB2:62:REAL[2]"); // 12.345, 12.345
-            builder.addTagAddress("lreal-value", "%DB2:70:LREAL"); // 3.14159265358979
-            builder.addTagAddress("lreal-array", "%DB2:78:LREAL[2]"); // 1.2345, -1.2345
+            
+//            builder.addTagAddress("lreal-value", "%DB2:70:LREAL"); // 3.14159265358979
```
Comment (chrisdutz) [inline plc4j/drivers/s7/src/test/java/org/apache/plc4x/java/s7/readwrite/DatatypesTest.java:67]: Why did you comment out these?

File: protocols/s7/src/main/resources/protocols/s7/s7.mspec [hunk 2]
```diff
@@ -897,6 +897,7 @@
     ['0x0B' UDINT          ['0x07'     , 'D'             , '4'               , 'INT'                 , 'INTEGER'                          , 'IEC61131_UDINT'          , 'false'             , 'false'             , 'true'               , 'true'               , 'true'              ]]
     ['0x0C' LINT           ['0x00'     , 'X'             , '8'               , 'INT'                 , 'BYTE_WORD_DWORD'                  , 'IEC61131_LINT'           , 'false'             , 'false'             , 'false'              , 'true'               , 'false'             ]]
     ['0x0D' ULINT          ['0x00'     , 'X'             , '8'               , 'INT'                 , 'BYTE_WORD_DWORD'                  , 'IEC61131_ULINT'          , 'false'             , 'false'             , 'false'              , 'true'               , 'false'             ]]
+    ['0x1C' COUNTER        ['0x1C'     , 'X'             , '2'               , 'null'                , 'OCTET_STRING'                  , 'IEC61131_INT'           , 'true'              , 'true'              , 'true'               , 'true'               , 'false'             ]]
```
Comment (chrisdutz) [inline protocols/s7/src/main/resources/protocols/s7/s7.mspec:900]: Hmmm it seems COUNTER has the same code as DATE_AND_TIME ... does that mean that support of COUNTER now breaks the ability to read DATE_AND_TIME?

Comment (glcj) [reply protocols/s7/src/main/resources/protocols/s7/s7.mspec:900]: Hi Chris,

Yes, but the enum type generated by the mspec does the magic.

Attached screenshot for a DATE_AND_TIME,

Best regards,



imagen

imagen

Comment (chrisdutz) [reply protocols/s7/src/main/resources/protocols/s7/s7.mspec:900]: Ok ... so I tested my Manual tests with the devices that I have and I didn't run into any problems. So please ignore this comment ;-)

**Pull request merged**